pagopa/io-app
# feat(Piattaforma Notifiche): [IAMVL-82] Handle attachments from PN message

## Short description
### ⚠️ **This PR depends on #4043** 

This branch adds the support for attachments in messages from PN. In order to reduce the cost of implementation, I've reused what we already did for MVL. I've extracted some functions from the MVL feature in order to use them also in PN, **however some implementation details should be refactored to make it a general solution (e.g. attachments are still stored in a `mvl` section)**. I suggest to address it on a specific task to be discussed later on.

To have a context for the current implementation, please refers to #3947.

https://user-images.githubusercontent.com/467098/179551431-bdcc5f6d-eeb0-4936-a7f7-88c1744dbba9.mp4

## How to test
Use the [IIP-18-add-pn-endpoints](https://github.com/pagopa/io-dev-api-server/tree/IIP-18-add-pn-endpoints) branch of the dev server and add this to your `config.json`:
```
{
  "messages": {
    "pnCount": 1
  },
  "services": {
    "includePn": true
  }
}
```
Set `PN_ENABLED=YES` to `.env` and run the app, then open a message from PN and try to download attachments.

When rendering fails (for example if the file downloaded is not a valid PDF) an error message should be displayed to the user, with the ability to open the downloaded file with another app.

https://user-images.githubusercontent.com/467098/179568963-2acd40a0-d079-4299-84d3-cd70d40246c9.mp4
